src/main.rs

```diff
@@ -120,7 +125,40 @@ fn main() {
 
     let build_server = remote.host;
 
-    let build_path = project_dir.clone();
+    let build_path = if args.use_temp_dir {
+        // Create a temporary directory on the remote server using mktemp
+        let project_name = project_dir.file_name().unwrap_or("unnamed-project");
+
+        info!("Creating temporary directory on remote server...");
+        let mktemp_output = Command::new("ssh")
+            .args(&["-p", &remote.ssh_port.to_string()])
+            .arg(&build_server)
+            .arg(format!("mktemp -d -t 'crunch-{}-XXXXXX'", project_name))
+            .output()
+            .unwrap_or_else(|e| {
+                error!(
+                    "Failed to create temporary directory on remote server (error: {})",
+                    e
+                );
+                exit(-3);
+            });
+
+        if !mktemp_output.status.success() {
+            error!("mktemp command failed on remote server");
+            error!("stderr: {}", String::from_utf8_lossy(&mktemp_output.stderr));
+            exit(-3);
+        }
+
+        let temp_path = String::from_utf8_lossy(&mktemp_output.stdout)
+            .trim()
+            .to_string();
+        info!("Created remote temporary directory: {}", temp_path);
+        temp_path
+    } else {
+        project_dir.to_string()
+    };
+
+    log::log!(log::Level::Info, "Using build path: {}", build_path);
```

Comment:
I think we could simplify this and avoid a RTT of latency by generating the uid locally instead of on the server:

```
let build_path = if args.use_temp_dir {
    let project_name = project_dir.file_name().unwrap_or("unnamed-project");
    let uid = /* generate uid */
    format!("/{}/{}-{}", remote.temp_dir, project_name, uid)
} else {
    project_dir.to_string()
};
```
Then we also don't need the `rsync_path_arg` logic later.